File: solutions/java/squeaky-clean/1/src/main/java/SqueakyClean.java
class SqueakyClean {
    static String clean(String identifier) {
        boolean camelWord = false;
        var strBld = new StringBuilder();

        for (int i = 0; i < identifier.length(); i++) {
            char ch = identifier.charAt(i);
            if (ch == ' ') {
                strBld.append('_');
            } else if (isLeetSpeakChar(ch)) {
                strBld.append(getLeetSpeakChar(ch));
            } else if (ch == '-') {
                camelWord = true;
            } else if (Character.isAlphabetic(ch)) {
                if (camelWord) {
                    ch = Character.toUpperCase(ch);
                    camelWord = false;
                }
                strBld.append(ch);
            }
        }
        return strBld.toString();
    }

    private static boolean isLeetSpeakChar(char ch) {
        return ch == '0' || ch == '1' || ch == '3' || ch == '4' || ch == '7';
    }

    private static char getLeetSpeakChar(char ch) {
        return switch (ch) {
            case '0' -> 'o';
            case '1' -> 'l';
            case '3' -> 'e';
            case '4' -> 'a';
            case '7' -> 't';
            default -> ' ';
        };
    }
}

File: solutions/csharp/squeaky-clean/1/SqueakyClean.cs
using System;
using System.Text;

public static class Identifier
{
  private static bool IsGreekLetterInRange(char ch)
  {
    return ch >= 'α' && ch <= 'ω';
  }

  private static bool IsCharKebabCase(string str, int charLocation)
  {
    return charLocation > 0 && str[charLocation - 1] == '-';
  }

  public static string Clean(string identifier)
  {
    StringBuilder stringBuilder = new StringBuilder();

    for (int i = 0; i < identifier.Length; i++)
    {
      char ch = identifier[i];

      if (Char.IsWhiteSpace(ch))
      {
        stringBuilder.Append("_");
      }
      else if (Char.IsControl(ch))
      {
        stringBuilder.Append("CTRL");
      }
      else if (Char.IsLetter(ch) && !IsGreekLetterInRange(ch))
      {
        if (IsCharKebabCase(identifier, i))
        {
          stringBuilder.Append(Char.ToUpper(ch));
        }
        else
        {
          stringBuilder.Append(ch);
        }
      }
    }

    return stringBuilder.ToString();
  }
}
